Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -2254,65 +2145,31 @@ RefPtr<CSSValue> consumeSingleWebkitBackgroundSize(CSSParserTokenRange& range, c
     return consumeBackgroundSize<CSSPropertyWebkitBackgroundSize>(range, context.mode);
 }
 
-bool isSelfPositionKeyword(CSSValueID id)
-{
-    return identMatches<CSSValueStart, CSSValueEnd, CSSValueCenter, CSSValueSelfStart, CSSValueSelfEnd, CSSValueFlexStart, CSSValueFlexEnd>(id);
-}
-
-bool isSelfPositionOrLeftOrRightKeyword(CSSValueID id)
+bool isFlexBasisIdent(CSSValueID id)
 {
-    return isSelfPositionKeyword(id) || isLeftOrRightKeyword(id);
+    switch (id) {
```

Comment:
How did you decide whether to use `identMatches` or not?

Reply:
In this case, I was merging in validWidthOrHeightKeyword(), since it was the only use, and it used a switch, so I just kept it that way. My feeling is that a switch is more natural and easier to read than identMatches<> for long lists, but don't have a strong preference.